includes/wf_crm_webform_postprocess.inc

```diff
@@ -1198,8 +1198,20 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
       // The api won't let us manually set status without this weird param
       $params['skipStatusCal'] = !empty($params['status_id']);
 
-      if (isset($this->ent['contribution_recur'][1]['id'])) {
-        $params['contribution_recur_id'] = $this->ent['contribution_recur'][1]['id'];
+      if (isset($this->ent['contribution_recur'][1]['id']) && !empty($params['auto_renew'])) {
+        $contributionRecurId = $this->ent['contribution_recur'][1]['id'];
+
+        $contributionRecur = civicrm_api3('ContributionRecur', 'get', [
+          'return' => 'auto_renew',
+          'sequential' => 1,
+          'id' => $contributionRecurId,
+        ]);
+
+        if (!empty($contributionRecur['values'][0]['auto_renew'])) {
+          $params['contribution_recur_id'] = $contributionRecurId;
+        }
+
+        unset($params['auto_renew']);
```

Comment:
@omarabuhussein What is this bit of code doing? Can we get a comment to explain it's purpose?

Reply:
If a membership is created to be autorenewed in civi core then two fields are affected : 

1- the recur contribution auto_renew field will be set to TRUE
2- the membership contribution_recur_id will be set to to refer to the created recur contribution.

The code above handles point 2 , will add a comment on it anyway.

Reply:
> I also ensure that only recur contributions with auto_renew field set to TRUE will set the contribution_recur_id field on the membership since any membership with this field set will appear on CiviCRM as an auto-renew membership.

The "auto_renew" field on the contribution_recur record is a bit confusing and depends on the processor implementation more than anything.  However, generally I think we should be setting the contribution_recur_id on the membership record *IF* the membership should be renewed when the recurring contribution receives another payment - that's what core will expect. So the membership may show "auto renew" even if the recurring contribution is "manually" renewed. If the contribution_recur_id is not set on the membership then the membership won't get renewed when the recurring contribution receives another payment.

Reply:
you are correct @mattwire, but just for safety I set both the auto_renew field on the contribution_recur record as well as the contribution_recur_id on the membership too.

Reply:
So @mattwire for the scenario of an annual membership paid by monthly DD instalments:  there is a recurring contribution that should be linked to the membership somehow, but the instalments should not extend the membership.  The membership may be set to autorenew, in which case the DD processor needs to recognise when the end of the membership period occurs, and extend the membership and update (or create a new) recurring contribution for the next 12 months.  How should that be configured?

Reply:
@aydun you are asking matt here but from my perspective there is no special configuration for such scenario, just at code level the membership contribution_recur_id should be set to the point to this recur contribution , and the recur contribution auto_renew field should be set to True (though according to Matt the auto_renew is not necessarily important since it more depends on the payment processor implementation where some make use of it and others don't ).
In this PR I am doing both, I am setting the contribution_recur_id on the membership to point to the recur contribution and I also set the auto_renew field on the recur contribution itself to True.

Reply:
> > the scenario of an annual membership paid by monthly DD instalments
@aydun The scenario you describe is not supported by CiviCRM core (though it would be nice to be able to).
If you link the membership to the recurring contribution it will renew the membership every time the recurring contribution receives a new payment.

> The membership may be set to autorenew

You can't set a membership to autorenew :-) You can link it to a recurring contribution at which point the UI shows "auto-renew" on the membership.